src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    public MyHashMap(int initialCapacity, float defaultLoadFactor) {
+        this.buckets = new Node[initialCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == buckets.length * DEFAULT_LOAD_FACTOR) {
+            resizeIfNeeded();
+        }
+        int index = (key == null) ? 0 : getIndex(key);
```

Comment:
```suggestion
        int index = getIndex(key);
```
You do not need this `(key == null) ? 0` check here as you have it inside `getIndex` method